### PR TITLE
gedmo: allow to switch content language on list and filter by language

### LINKED ISSUE #81: Translation on list
Hello,

You recently added the translation for the object show. 

I need the same on the object list. 

I also need the filters to use the translation on list, filter on every locale available...

Thanks.

### PR DESCRIPTION
**Pull request by dbu**
fix #81, fix #83

continuation of #83

this can be merged after 1.0.2 has been tagged - its a new feature.

### CODE REVIEW

**Comment by dbu** [comment]
@mcrinquand something with rebasing seems to have gone wrong. i took your branch and cleaned it up. i also added that this will bump the version to 1.1 as its a new feature. i wonder if we might need an option to disable this feature... but well, if somebody needs to disable it, they could do a PR to add that option...

i will wait with merging until i tagged 1.0.2 which hopefully will happen soon.

File: Admin/Extension/Gedmo/TranslatableAdminExtension.php [hunk 1]
```diff
@@ -62,4 +64,61 @@ public function alterObject(AdminInterface $admin, $object)
...
+        foreach ($joinDqlParts as $joins) {
+            foreach ($joins as $join) {
+                if ($join->getAlias() === 't') {
+                    $aliasAlreadyExists = true;
+                    break 2;
+                }
+            }
+        }
+
+        if ($aliasAlreadyExists === false) {
```
Comment (greg0ire) [inline Admin/Extension/Gedmo/TranslatableAdminExtension.php]: `$aliasAlreadyExists` is a boolean, I think you can use it directly in the condition, which is what booleans are made for, so `if (!$aliasAlreadyExists)`, which sounds closer to plain english

File: Resources/views/Block/block_locale_switcher.html.twig [hunk 2]
```diff
@@ -1,8 +1,15 @@
 {% set object = block_context.settings.object %}
 {% set admin  = block_context.settings.admin %}
 {% set locale_switcher_route = block_context.settings.locale_switcher_route %}
+{% set currentLocale = null %}
+
+{% if (admin.class is translatable) %}
+    {% for extension in admin.extensions %}
+        {% if(extension.translatableLocale is defined) %}
```
Comment (greg0ire) [inline Resources/views/Block/block_locale_switcher.html.twig]: missing space after the `if`

File: Admin/Extension/Gedmo/TranslatableAdminExtension.php [hunk 3]
```diff
@@ -62,4 +64,61 @@ public function alterObject(AdminInterface $admin, $object)
...
+     * To use with a doctrine_orm_callback filter type.
+     *
+     * @param ProxyQuery $queryBuilder
+     * @param string     $alias
+     * @param string     $field
+     * @param string     $value
+     *
+     * @return bool
+     */
+    public static function translationFieldFilter(ProxyQuery $queryBuilder, $alias, $field, $value)
```
Comment (dbu) [inline Admin/Extension/Gedmo/TranslatableAdminExtension.php]: looking at the code this seems to never be used. if the user needs to do something to activate the filter, can you please add some documentation for this @mcrinquand ?

File: Admin/Extension/Gedmo/TranslatableAdminExtension.php [hunk 4]
```diff
@@ -62,4 +64,61 @@ public function alterObject(AdminInterface $admin, $object)
...
+     * To use with a doctrine_orm_callback filter type.
+     *
+     * @param ProxyQuery $queryBuilder
+     * @param string     $alias
+     * @param string     $field
+     * @param string     $value
+     *
+     * @return bool|null
+     */
+    public static function translationFieldFilter(ProxyQuery $queryBuilder, $alias, $field, $value)
```
Comment (dbu) [inline Admin/Extension/Gedmo/TranslatableAdminExtension.php]: looking at the code this seems to never be used. if the user needs to do something to activate the filter, can you please add some documentation for this @mcrinquand ?

**Comment by mcrinquand** [comment]
Yes @dbu I'll update the documentation.

**Comment by dbu** [comment]
thanks. can you take this branch translate-gedmo-lists and do the pull request against it instead of master? that way we can keep the changes together and i can already merge the doc update.

**Comment by dbu** [comment]
ping @mcrinquand

**Comment by Soullivaneuh** [comment]
@dbu The PR should be re-opened against master.

Could you also please do your work from a fork? Working directly on the repo would be a problem with StyleCI sending auto fix PR.

I'm closing this one.

Thanks.

**Comment by dbu** [comment]
@mcrinquand any chance you can pick this up and finalize it?

File: Admin/Extension/Gedmo/TranslatableAdminExtension.php [hunk 5]
```diff
@@ -62,4 +64,61 @@ public function alterObject(AdminInterface $admin, $object)
...
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureQuery(AdminInterface $admin, ProxyQueryInterface $query, $context = 'list')
+    {
+        $this->getTranslatableListener($admin)->setTranslatableLocale($this->getTranslatableLocale($admin));
+        $this->getTranslatableListener($admin)->setTranslationFallback('');
```
Comment (Soullivaneuh) [inline Admin/Extension/Gedmo/TranslatableAdminExtension.php]: From Scrutinizer:

> `''` is of type `string`, but the function expects a `boolean`.

**Pull request closed without merging**